src/main/java/at/hannibal2/skyhanni/features/inventory/ShiftClickNPCSell.kt

```diff
@@ -0,0 +1,53 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class ShiftClickNPCSell {
+
+    val config get() = SkyHanniMod.feature.inventory.shiftClickNPCSell
+
+    val sellSlot = 49
+    val lastLoreLineOfSellPattern by RepoPattern.pattern(
+        "inventory.npc.sell.lore",
+        "§7them to this Shop!|§eClick to buyback!"
+    )
+
+    var isInNPCSell = false
+
+    fun enabled() = LorenzUtils.inSkyBlock && config
+
+    @SubscribeEvent
+    fun onOpen(event: InventoryFullyOpenedEvent) {
+        if (!enabled()) return
+        isInNPCSell = lastLoreLineOfSellPattern.matches(event.inventoryItems[sellSlot]?.getLore()?.lastOrNull() ?: "")
```

Comment:
scrap the other one, mine is better (performance and for maintaining it)

Reply:
Then please make this class to an object, inInventory public, use it in hide non clickable items, and remove the unused fields (but not the stuff in repo, otherwise older versions break)

Reply:
done, but when reworking the hideNotClickableItems this should be changed